tools/rule_extractor/rule_extractor.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+
+'''
+Utility module to read and write Prometheus Alert rules to file.
+Ensures we can test the Alert rules deployed and for documentation
+purposes.
+'''
+
+import argparse
+import functools
+import ipaddress
+import json
+import operator
+import requests
+import subprocess
+import sys
+import yaml
+
+
+class AlertRuleFile:
+    """A class for manipulating Prometheus Alert & Recording Rules."""
+
+    def __init__(self, ip, port):
+        self.ip = ip
+        self.port = port
+
+    def query_prometheus_api(self, route=False):
+        """Read the Prometheus route contents into memory."""
+        if not route:
+            raise Exception("Expected endpoint route but got None")
+
+        endpoint = "https://{}:{}/api/prometheus/api/v1/{}".format(
+            self.ip, self.port, route
+        )
+        try:
+            response = requests.get(url=endpoint, verify=False)
+        except requests.exceptions.ConnectionError as exc:
+            raise Exception("Failed to reach endpoint {}: {!s}".format(
+                route, exc
+                )
+            )
+        if response.status_code != 200:
+            raise Exception("Expected status code <200> but got {}".format(
+                response.status_code
+                )
+            )
+
+        return response.json()
+
+    def write_to_file(self, content, rule_name):
+        with open('{}.json'.format(rule_name), 'w', encoding='utf-8') as f:
```

Comment:
So, files are generated in the current working directory, not in `tools/rule_extractor/`, I first need to `cd` into this directory to launch this script and update the committed JSON files.